### PR TITLE
PWGHF: Add Heavy Flavour electron Hadron correlation 

### PR DESCRIPTION
**Pull request by rashigupt**
Heavy Flavour Electron Hadron Correlation table and task

### CODE REVIEW

File: PWGHF/HFC/TableProducer/correlatorHfeHadrons.cxx [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file CorrelatorHfeHadrons.cxx
```
Comment (vkucera) [inline PWGHF/HFC/TableProducer/correlatorHfeHadrons.cxx:12]:
```suggestion
/// \file correlatorHfeHadrons.cxx
```

File: PWGHF/HFC/TableProducer/correlatorHfeHadrons.cxx [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+                    McTableTracks const& mcTracks)
+  {
+    fillCorrelation(mcCollision, mcElectron, mcTracks);
+  }
+
+  PROCESS_SWITCH(HfCorrelationHfeHadrons, processMcRec, "Process MC Reco mode", false);
+  // Processing Event Mixing
+  Preslice<aod::Tracks> perCol = aod::track::collisionId;
+  Preslice<aod::HfSelEl> perCollision = aod::hf_sel_electron::collisionId;
+  void processDataMixedEvent(TableCollisions const& collision, aod::HfSelEl const& electron, TableTracks const& tracks)
```
Comment (vkucera) [inline PWGHF/HFC/TableProducer/correlatorHfeHadrons.cxx:215]: Do not declare data members in between methods. 
```suggestion
  PROCESS_SWITCH(HfCorrelationHfeHadrons, processMcRec, "Process MC Reco mode", false);

  void processDataMixedEvent(TableCollisions const& collision, aod::HfSelEl const& electron, TableTracks const& tracks)
```

**Comment by rashigupt** [comment]
Please check it and provide your suggestions.

File: PWGHF/HFC/Tasks/taskCorrelationHfeHadrons.cxx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file taskCorrelationHfeHadrons.cxx
+/// \brief HFE-Hadrons azimuthal correlations analysis task - data-like, MC-reco and MC-Gen analyses
+/// \author Rashi Gupta <rashi.gupta@cern.ch>, IIT Indore
+/// \author Ravindra Singh <ravindra.singh@cern.ch>, IIT Indore
+
+#include "THnSparse.h"
```
[the remark this answers is not in the document]
Comment (vkucera) [reply PWGHF/HFC/Tasks/taskCorrelationHfeHadrons.cxx]: Thanks

**Review by vkucera — APPROVED** [review]

**Review by deepathoms — APPROVED** [review]

**Pull request merged**